include/clang/AST/OperationKinds.def

```diff
@@ -327,6 +327,10 @@ CAST_OPERATION(AddressSpaceConversion)
 // Convert an integer initializer to an OpenCL sampler.
 CAST_OPERATION(IntToOCLSampler)
 
+// CK_UnCheckedToChecked - A conversion from a Unchecked pointer to checkedc
+// pointer
+CAST_OPERATION(UnCheckedToChecked)
```

Comment:
This should be renamed to something else, such as PointerBounds.   The source operand could actually be a checked pointer type.  We might be casting from array_ptr<char> to array_ptr<int>, for example.